nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -169,13 +172,20 @@ public Nullness valueOfSubNode(Node node) {
     @Override
     public NullnessStore<Nullness> initialStore(
             UnderlyingAST underlyingAST, List<LocalVariableNode> parameters) {
+        lambdaResultExpression = null;
         if (parameters == null) {
             // Documentation of this method states, "parameters is only set if the underlying AST is a
             // method"
             return NullnessStore.empty();
         }
         boolean isLambda = underlyingAST.getKind().equals(UnderlyingAST.Kind.LAMBDA);
         if (isLambda) {
+            LambdaExpressionTree lambdaTree = ((UnderlyingAST.CFGLambda) underlyingAST).getLambdaTree();
+            if (lambdaTree.getBodyKind().equals(LambdaExpressionTree.BodyKind.EXPRESSION)) {
+                // If this is a lambda with a single expression as its body, then remember that expression while
+                // visiting the code inside the lambda, so we can call handler.onDataflowVisitLambdaResultExpression
+                lambdaResultExpression = (ExpressionTree) lambdaTree.getBody();
```

Comment:
Is it possible to have nested lambdas such that this needs to be a stack?

Reply:
Yes, this is possible. I made the change and, more importantly, added a test case to catch it. But consider this still a WIP until we have figured out if there is a cleaner way to do this (if there isn't, then I at least need to go over the `visitX` methods of this class, one by one, and make sure we are testing for any that might be a boolean expression).